lib/govuk_message_queue_consumer/rabbitmq_config.rb

```diff
@@ -10,7 +10,24 @@ def self.from_environment(env)
         user: fetch(env, "RABBITMQ_USER"),
         pass: fetch(env, "RABBITMQ_PASSWORD"),
         recover_from_connection_close: true,
-      }
+      }.merge(tls_config(env))
+    end
+
+    def self.tls_config(env)
+      # TLS config is optional - local RabbitMQs won't use TLS,
+      # but Amazon MQ enforces it. So we default to non-TLS.
+      if env.fetch("RABBITMQ_TLS", false)
+        {
+          tls: true,
+          tls_cert: env.fetch("RABBITMQ_TLS_CERT", nil),
+          tls_key: env.fetch("RABBITMQ_TLS_KEY", nil),
```

Comment:
I'm probably over thinking this but I'm wondering if the RABBITMQ prefix is a bit misleading since I guess this the TLS info of the client and that RabbitMQ actually has it's own certificate?

I'm not too sure of a better prefix though - perhaps BUNNY_ since this is config for the Bunny gem rather than RABBITMQ itself?

Reply:
Since Bunny already uses the `RABBITMQ_` prefix as you mention, I think it's best not to introduce a `BUNNY_` prefix. I'd suggest `RABBITMQ_CLIENT_TLS_KEY` etc. to make it clear that this is the client cert/key.

Zooming out a bit, are we 100% sure we need to configure any of this TLS stuff? We don't need client cert authentication and it doesn't look like Amazon requires it. Bunny [uses OpenSSL's CA cert store by default](https://github.com/ruby-amqp/bunny/blob/ed26b1e9/lib/bunny.rb#L71), for verifying the server cert. So I think there's a good chance we can just put `amqps` in `RABBITMQ_URL` and dispense with `GovukMessageQueueConsumer::RabbitMQConfig` altogether.

Reply:
Yep, that's one of the things we're going to be testing in integration over the next week or so (hopefully).